storage/storage_interface.ml

```diff
@@ -100,441 +186,1076 @@ let parse_nbd_uri nbd =
   | _ -> fail ()
 
 (** Uniquely identifies the contents of a VDI *)
-type content_id = string
+type content_id = string [@@deriving rpcty]
+
 
 (** The result of an operation which creates or examines a VDI *)
 type vdi_info = {
-    vdi: vdi;
-    uuid: string option;
-	content_id: content_id;
-    name_label: string;
-    name_description: string;
-    ty: string;
-    (* sm_config: workaround via XenAPI *)
-    metadata_of_pool: string;
-    is_a_snapshot: bool;
-    snapshot_time: string;
-    snapshot_of: vdi;
-    (* managed: workaround via XenAPI *)
-    read_only: bool;
-    cbt_enabled: bool;
-    (* missing: workaround via XenAPI *)
-    virtual_size: int64;
-    physical_utilisation: int64;
-    (* xenstore_data: workaround via XenAPI *)
-	persistent: bool;
-    sharable: bool;
-    sm_config: (string * string) list;
-}
-
-let default_vdi_info = {
-    vdi = "";
-    uuid = None;
-    content_id = "";
-    name_label = "";
-    name_description = "";
-    ty = "user";
-    metadata_of_pool = "";
-    is_a_snapshot = false;
-    snapshot_time = Xapi_stdext_date.Date.to_string Xapi_stdext_date.Date.never;
-    snapshot_of = "";
-    read_only = false;
-    cbt_enabled = false;
-    virtual_size = 0L;
-    physical_utilisation = 0L;
-    persistent = true;
-    sharable = false;
-    sm_config = [];
-}
-
-let vdi_info_of_rpc rpc = Rpc.struct_extend rpc (rpc_of_vdi_info default_vdi_info) |> vdi_info_of_rpc
-
-type sr_health = Healthy | Recovering
+  vdi: Vdi.t;
+  uuid: string option [@default None];
```

Comment:
Isn't the default always None, or did you just want to be explicit?

Reply:
Better to be explicit I think - but you're right, omitting it would do the same thing when unmarshalling. However, putting a default in like this allows us to put it in the docs later too.